updater/helpers.py

```diff
@@ -24,3 +26,23 @@ def executeCommand(command, stdin = None, cwd = None):
 # Convenience function to parse a date from a string
 def parseDate(string):
 	return datetime.datetime.strptime(string, "%Y-%m-%d").date()
+
+# The location of the local working copy of the data repository
+def locateDataDirectory():
+	tmpDirectory = configuration["tmpDirectory"]
+	return os.path.join(tmpDirectory, "hubble-data")
+
+# Create a local clone of the data repository if not present and update it
+def prepareDataDirectory(dataDirectory):
+	# Create data directory if not existing
+	if not os.path.exists(dataDirectory):
+		os.makedirs(dataDirectory)
+
+	# Get the data directory up-to-date
+	if not configuration["dryRun"]:
```

Comment:
That is side effect might surprise a caller of this function. I would move this check outside of the function.

Reply:
True, but even during a dry run we need the data directory to be created. This is because the data is still written to the TSV files but just not committed with Git.

Placing the check outside the function would prevent the data directory from being created if necessary, which would lead to errors.

As an alternative, we’d have to split the `prepareDataDirectory` function into two separate functions, one creating the directory if necessary and the other one updating the data to the latest version.

Reply:
Would it make sense to pass a parameter `dryrun` to the function?

Reply:
or call it `perform_clone`?

Reply:
Yep, that’s a good suggestion. I’ll make the change!